### PR TITLE
Add validation for resource functions with return type error

### PR DESCRIPTION
**Pull request by aashikam**
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1294

## Purpose
$title

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/PluginConstants.java [hunk 1]
```diff
@@ -53,7 +53,9 @@
         INVALID_MULTIPLE_LISTENERS("Multiple listener attachments. Only one graphql:Listener is allowed.",
                 "GRAPHQL_107", DiagnosticSeverity.ERROR),
         INVALID_MAX_QUERY_DEPTH("Invalid maxQueryDepth value. Value must be a positive integer",
-                "GRAPHQL_108", DiagnosticSeverity.ERROR);
+                "GRAPHQL_108", DiagnosticSeverity.ERROR),
+        INVALID_RETURN_TYPE_ERROR("Invalid return type error. Resource function must have a return data type.",
+                "GRAPHQL_109", DiagnosticSeverity.ERROR);
```
Comment (shafreenAnfar) [inline graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/PluginConstants.java:58]: This is also incorrect right ?

```ballerina
isolated resource function get greeting() returns error|string
```

File: graphql-compiler-plugin-tests/src/test/resources/ballerina_sources/invalid_service_9/service.bal [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/graphql;
+
+service graphql:Service on new graphql:Listener(4000) {
+    isolated resource function get greeting() returns error {
```
Comment (shafreenAnfar) [inline graphql-compiler-plugin-tests/src/test/resources/ballerina_sources/invalid_service_9/service.bal:20]: This is also incorrect right ?

```ballerina
isolated resource function get greeting() returns error|string
```

Comment (ThisaruGuruge) [reply graphql-compiler-plugin-tests/src/test/resources/ballerina_sources/invalid_service_9/service.bal:20]: This should be valid.

**Review by ThisaruGuruge — APPROVED** [review]
LGTM

**Comment by ThisaruGuruge** [comment]
This looks fine, but we have to validate all the service classes returned from GraphQL resource functions. 
Created an issue for this: https://github.com/ballerina-platform/ballerina-standard-library/issues/1295

**Pull request merged**